Sources/NIOFileSystem/Internal/System Calls/Syscall.swift

```diff
@@ -347,7 +347,11 @@ public enum Libc {
         return valueOrErrno {
             pathBytes.withUnsafeMutableBufferPointer { pointer in
                 // The array must be terminated with a nil.
+                #if os(Android)
+                libc_fts_open([pointer.baseAddress!, unsafeBitCast(0, to: UnsafeMutablePointer<CInterop.PlatformChar>.self)], options.rawValue)
```

Comment:
Is `unsafeBitCast(0, to: UnsafeMutablePointer<CInterop.PlatformChar>.self)` any different to `UnsafeMutablePointer(bitPattern: 0)`?

Reply:
The Swift compiler gave me the same recommendation weeks ago:
```
swift-nio/Sources/NIOFileSystem/Internal/System Calls/Syscall.swift:351:54: warning: 'unsafeBitCast' from 'Int' to 'UnsafeMutablePointer<CInterop.PlatformChar>' (aka 'UnsafeMutablePointer<Int8>') can be replaced with 'bitPattern:' initializer on 'UnsafeMutablePointer<CInterop.PlatformChar>' (aka 'UnsafeMutablePointer<Int8>')
                libc_fts_open([pointer.baseAddress!, unsafeBitCast(0, to: UnsafeMutablePointer<CInterop.PlatformChar>.self)], options.rawValue)
                                                     ^~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                     UnsafeMutablePointer<CInterop.PlatformChar>(bitPattern:  )
```
so I tried it, replacing this line with
```
libc_fts_open([pointer.baseAddress!,  UnsafeMutablePointer<CInterop.PlatformChar>(bitPattern: 0)], options.rawValue)
```
That was a no go for compilation:
```
swift-nio/Sources/NIOFileSystem/Internal/System Calls/Syscall.swift:351:55: error: cannot convert value of type 'UnsafeMutablePointer<CInterop.PlatformChar>?' (aka 'Optional<UnsafeMutablePointer<Int8>>') to expected element type 'UnsafeMutablePointer<CInterop.PlatformChar>' (aka 'UnsafeMutablePointer<Int8>')
                libc_fts_open([pointer.baseAddress!,  UnsafeMutablePointer<CInterop.PlatformChar>(bitPattern: 0)], options.rawValue)
                                                      ^
```
OK, so let's try force-unwrapping instead with this line and see what happens:
```
libc_fts_open([pointer.baseAddress!,  UnsafeMutablePointer<CInterop.PlatformChar>(bitPattern: 0)!], options.rawValue)
```
As might be expected, tests now fail at runtime:
```
NIOFileSystem/Syscall.swift:351: Fatal error: Unexpectedly found nil while unwrapping an Optional value
```
That's why I used this bitcast instead.